Replace submitted note table colors with created/resolved

AntonKhorev opened the pull request:
Instead of *submitted* and *commented* note colors switch to *created*, *resolved* and *commented*.

Reasons:
- Close actions were colored as *commented* even if notes were closed without any comments. Now they are going to be colored as *resolved*.
- *Submitted* is not a term we use for notes, we have "add a note" and notes "created by".
- There was a [complaint](https://github.com/openstreetmap/openstreetmap-website/pull/4700#issuecomment-2221682594) about the primary color used for submitted notes that is blue making links harder to see. What other color can we use? For example, the *success* color, like we do for unread messages. But that's the color of the resolved note marker, and then we might as well add the unresolved color.

Caveats:
- I keep "submitted" in locale keys to avoid errors while `subheading_html` is not updated.
- I ignore *reopen* events because if I don't, I'd have to treat them as *open* events but then there wouldn't be an equivalent to the previous *submitted* state. And the equivalent is green *created* + yellow *created and resolved*.
- I had to add this *created and resolved* state because what if the user did both actions? I can't pick one more important color. If I look at note status changes, the latest action is the most important one and I'd have to color notes as *resolved*. But then again I'd lose the equivalent to the previous *submitted* state.

![image](https://github.com/user-attachments/assets/03ac5999-0822-4c03-938d-a8447f3114a4)

Dark mode:
![image](https://github.com/user-attachments/assets/8c88a294-bb1a-42d5-b169-051f68f34916)

Legend while locale strings aren't updated:
![image](https://github.com/user-attachments/assets/754fcbe2-f27d-4103-9359-71932981a56a)

tomhughes commented:
You say you ignore reopen events but in fact this seems to extend the concept of submitted/created from who initially opened it to include anybody that reopens it as well? You are considered to have created it if there are any open events by you.

Obviously that is mirrored in terms of closing by looking for any close event, but aren't users going to be more interested in whether the note is currently open or closed and whether this person closed it?

tomhughes commented:
After your change the test is:

```ruby
opened_by_user = note.comments.any? { |comment| comment.author == @user && comment.event == "opened" }
```

which will be true if the user has even opened it, either the original open or a reopen?

AntonKhorev commented:
This is a test only for original open events:

> comment.event == "opened"

Reopen events are stored as `"reopened"`.

AntonKhorev commented:
Another option is to ignore all close events other than the last one, that's symmetrical to ignoring reopen events.

tomhughes commented:
Oh sorry I looked at the source last night and was quite sure they were all marked as opened.

tomhughes commented:
Why check every comment though when the initial open will always be the first one?

There's still a question about resolution I think? Should we be looking at every one or just the last one?

AntonKhorev commented:
There are some drawbacks in looking only at the last close event.

#5255 tried to do filtering by interaction type but the db queries are ugly. To optimize them we could have a users x notes table where we store among other things interaction flags. When a user closed a note, we set a closed flag there, then we can filter notes by that flag. But if we want to pay attention only to the last close event, we'd have to clear that flag when somebody reopens the note.

tomhughes commented:
My thinking was literally just to look at the last event which should be the close event unless it's been reopened and if it has then do we want to consider it closed by this user?

AntonKhorev commented:
Alright, let's not think about filtering for now. Changed to look only at the last close event.

I also wanted the "commented" indication not to be misleading. It includes more than commenting and this change stretches it further by including non-last close events. So I removed the claim that not colored notes were commented, although the sentence gets a bit long:

![image](https://github.com/user-attachments/assets/ca37df6c-4920-4adb-abe4-d1a638e378c9)

mmd-osm commented:
<!--
  1 failure:  `bla.sh` failed.
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  `bla.sh` failed.
  </td>
    </tr>
  </tbody>
</table>



### Log

<details>


```

```


</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7aeafbd890108734b866df1eaac8ef9b595d5cab
</p>

mmd-osm commented:
<!--
  1 failure:  `bla.rb` failed.
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  `bla.rb` failed.
  </td>
    </tr>
  </tbody>
</table>



### Log

<details>


```
Hello from ruby!
{
  "fails": [
    {
      "message": "Gem/Bundle changes were not documented in the Description. If this is a major update, add #gemlock_major_upgrade to the PR Description."
    }
  ],
  "warnings": [],
  "messages": [],
  "markdowns": [],
  "scheduled": []
}

```


</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7aeafbd890108734b866df1eaac8ef9b595d5cab
</p>

mmd-osm commented:
<!--
  1 failure:  bla
  0 warning: 
  
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>bla</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7aeafbd890108734b866df1eaac8ef9b595d5cab
</p>

tomhughes commented:
Can you please stop abusing this PR for testing something that we're never likely to use! Just give me time to figure out the problems with the ruby version.

mmd-osm commented:
<!--
  1 failure:  Gem/Bundle change...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Gem/Bundle changes were not documented in the Description. If this is a major update, add #gemlock_major_upgrade to the PR Description.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7aeafbd890108734b866df1eaac8ef9b595d5cab
</p>

mmd-osm commented:
<!--
  1 failure:  Gem/Bundle change...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Gem/Bundle changes were not documented in the Description. If this is a major update, add #gemlock_major_upgrade to the PR Description.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7aeafbd890108734b866df1eaac8ef9b595d5cab
</p>

nertc reviewed (commented):
Tested and works perfectly for all kinds of situations. I like the last solution that UI watches only author and the last comment. It provides intuitive UI/UX.

AntonKhorev commented on app/views/notes/index.html.erb:
```diff
@@ -24,7 +26,15 @@
   <% @notes.each do |note| -%>
-    <tr<% if note.author == @user %> class="table-primary"<% end %>>
+    <% opened_by_user = note.author == @user %>
+    <% closed_by_user = note.comments.last&.author == @user && note.comments.last&.event == "closed" %>
```
Moved logic to `note_list_row_class` helper.
